lambdas/package/index.py

```diff
@@ -0,0 +1,144 @@
+"""
+Provide the head of a (potentially gzipped) file in S3. Stream to limit
+disk and RAM pressure.
+
+Lambda functions can have up to 3GB of RAM and only 512MB of disk.
+"""
+
+import io
+
+import boto3
+import pandas as pd
+
+from t4_lambda_shared.decorator import api, validate
+from t4_lambda_shared.utils import get_default_origins, make_json_response
+
+S3_DOMAIN_SUFFIX = '.amazonaws.com'
+
+SCHEMA = {
+    'type': 'object',
+    'properties': {
+        'bucket': {
+            'type': 'string'
+        },
+        'key': {
+            'type': 'string'
+        },
+        'access_key': {
+            'type': 'string'
+        },
+        'secret_key': {
+            'type': 'string'
+        },
+        'prefix': {
+            'type': 'string'
+        },
+    },
+    'required': ['bucket', 'key', 'access_key', 'secret_key'],
+    'additionalProperties': False
+}
+
+
+def load_df(s3response):
+    """
+    Read a streaming response from s3 select into a
+    Pandas DataFrame
+    """
+    buffer = io.StringIO()
+    for event in s3response['Payload']:
+        if 'Records' in event:
```

Comment:
It looks like we have to check that `End` is received: https://botocore.amazonaws.com/v1/documentation/api/latest/reference/eventstream.html